src/test/java/scmspain/karyon/restrouter/core/ParamTypeResolverTest.java

```diff
@@ -0,0 +1,37 @@
+package scmspain.karyon.restrouter.core;
+
+import com.tngtech.java.junit.dataprovider.DataProvider;
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import com.tngtech.java.junit.dataprovider.UseDataProvider;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.*;
+
+@RunWith(DataProviderRunner.class)
+public class ParamTypeResolverTest<T> {
+
+  @DataProvider
+  public static Object[][] dataProviderAdd() {
+    // @formatter:off
+    return new Object[][] {
+        { String.class, "lorem ipsum" },
+        { Boolean.class, "true" },
+        { Long.class, "27" },
+        { Integer.class, "42" },
+        { Short.class, "2" },
+        { Float.class, "27.42" },
+        { Double.class, "42.42" }
+
```

Comment:
I'd also add some negative test cases in order to document the behavior when somebody call Resolver with things like 
```
SomeExoticClass.class, "whatever value"
```